catalog/transformer/catalog-transformer-csv-common/src/test/java/ddf/catalog/transformer/csv/common/CsvTransformerTest.java

```diff
@@ -0,0 +1,223 @@
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.junit.Before;
+import org.junit.Test;
+
+public class CsvTransformerTest {
+
+  private static final List<AttributeDescriptor> ATTRIBUTE_DESCRIPTOR_LIST = new ArrayList<>();
+  private static Map<String, Attribute> metacardDataMap = new HashMap<>();
+
+  private static List<Metacard> metacardList = new ArrayList<>();
+  private static final int METACARD_COUNT = 2;
+
+  private static final String CSV_REGEX = "[\\n\\r,]";
+
+  private static final Object[][] ATTRIBUTE_DATA = {
+    {"attribute1", "value1", BasicTypes.STRING_TYPE},
+    {"attribute2", "value2", BasicTypes.STRING_TYPE},
+    {"attribute3", 101, BasicTypes.INTEGER_TYPE},
+    {"attribute4", 3.14159, BasicTypes.DOUBLE_TYPE},
+    {"attribute5", "value,5", BasicTypes.STRING_TYPE},
+    {"attribute6", "value6", BasicTypes.STRING_TYPE},
+    {"attribute7", "OBJECT", BasicTypes.OBJECT_TYPE},
+    {"attribute8", "BINARY", BasicTypes.BINARY_TYPE}
+  };
+
+  @Before
+  public void setup() {
+    ATTRIBUTE_DESCRIPTOR_LIST.clear();
+    metacardList.clear();
+    buildMetacardDataMap();
+    buildMetacardList();
+  }
+
+  @Test
+  public void getAllAttributes() {
+    Set<String> hiddenFields = new HashSet<>();
+    hiddenFields.add("attribute1");
+
+    Set<AttributeDescriptor> allAttributes =
+        CsvTransformer.getAllAttributes(metacardList, hiddenFields);
+    assertThat(allAttributes, hasSize(5));
+
+    Set<String> allAttributeNames =
+        allAttributes.stream().map(AttributeDescriptor::getName).collect(Collectors.toSet());
+
+    final Set<String> expectedAttributes =
+        Sets.newHashSet("attribute2", "attribute3", "attribute4", "attribute5", "attribute6");
+
+    assertThat(allAttributeNames, is(expectedAttributes));
+  }
+
+  @Test
+  public void getOnlyRequestedAttributes() {
+    Set<String> requestedAttributes = new HashSet<>();
+    requestedAttributes.add("attribute1");
+
+    Set<AttributeDescriptor> onlyRequestedAttributes =
+        CsvTransformer.getOnlyRequestedAttributes(
+            metacardList, requestedAttributes, Collections.emptySet());
+
+    assertThat(onlyRequestedAttributes, hasSize(1));
+    final String name = onlyRequestedAttributes.stream().findFirst().get().getName();
+    assertThat(name, is("attribute1"));
+  }
+
+  @Test
+  public void getOnlyRequestedAttributesWithExcludedConfig() {
+    Set<String> requestedAttributes = Sets.newHashSet("attribute1", "attribute2");
+    Set<String> excludedAttributes = Sets.newHashSet("attribute1");
+
+    Set<AttributeDescriptor> onlyRequestedAttributes =
+        CsvTransformer.getOnlyRequestedAttributes(
+            metacardList, requestedAttributes, excludedAttributes);
+
+    assertThat(onlyRequestedAttributes, hasSize(1));
+    final String name = onlyRequestedAttributes.stream().findFirst().get().getName();
```

Comment:
This should make an assertion that the result of `findFirst()` is present before calling `get()` and checking the value inside.